### ISSUE TITLE
Cannot install hooks for a submodule due to wrong path

### ISSUE THREAD

**Issue by extmind** [Arkweid/lefthook#104]
I've tried to use lefthook in the checked-out tree of a submodule. The directory structure is this:
```
/home/user/
-- project/          -- git repository
   -- shared/        -- subdirectory
      -- submodule/  -- submodule
```

I've placed a `lefthook.yml` file in the submodule's directory and ran `lefthook install` there. Here's the error message:
```
% lefthook install
SYNCING lefthook.yml
SERVED HOOKS: pre-commit, prepare-commit-msg
open /home/user/project/shared/submodule/home/user/project/.git/modules/shared/submodule/hooks/pre-commit: no such file or directory
```
The path isn't completely wrong, but its prefixed with the submodule's directory. Thus no hooks were installed at `/home/user/project/.git/modules/shared/submodule/hooks/`.

I'm using the 0.7.0 release. (Btw. `lefthook version` only reported that from inside a git working tree and just said `exit status 128` otherwise.)

**Comment by Arkweid** [comment]
@jsmestad have added submodules feature. Any idea what went wrong here?

**Comment by jsmestad** [comment]
@Arkweid the feature only runs things in relative paths from the root. What @extmind is talking about sounds like they want to a put the lefthook config somewhere relative to the git root (which should not have changed from the original design -- whatever that was ?)

I am unclear when you say submodule, are you referring to git submodules or just a relative path from root?

**Comment by Arkweid** [comment]
@extmind by design lefthook should be installed in git top-level repository. If you want to change this behavior, PR is welcome as always.

**Comment by extmind** [comment]
> I am unclear when you say submodule, are you referring to git submodules or just a relative path from root?

Yes, I'm referring to git submodules. The `project/shared/submodule` directory is a repository working-tree, created via the `git submodule` cli, it has a `.git` file referring to its real `.git` at `project/.git/modules/shared/submodule/`.

So, I've tried looking into the code despite not knowing much about golang. It seems that lefthook determines the root of an repository's working-tree by calling `git rev-parse --show-toplevel`, which when in some directory below (or exactly in) `project/shared/submodule` refers to this directory itself. This is, in my opinion, correct, because a submodule is for itself a fully-fledged git repository (just with a slightly different `.git` location). This also means that it can find the `lefthook.yml` file in the submodule's directory.
On adding a hook, lefthook determines the `.git` and in turn the `.git/hooks` directory by calling `git rev-parse --git-common-dir`. When called from inside the submodule this yields an *absolute* path: `/home/user/project/.git/modules/shared/submodule`, but outside it yiels a *relative* path like e.g. `../.git`. Note that `git rev-parse --git-dir` and `git rev-parse --absolute-git-dir` exist, but there is no `git rev-parse --absolute-git-common-dir`. But I'm sure golang would have some function to convert these itself.
Because lefthook assumes, that `git rev-parse --git-common-dir` always returns a relative path, it breaks in the submodule case and might do so in other cases as well (one would need to look at the git source code to determine that).


> @extmind by design lefthook should be installed in git top-level repository. If you want to change this behavior, PR is welcome as always.

As I wrote, I'm not a golang programmer. I've tested a small patch against the current release, switching `--git-common-dir` to `--absolute-git-dir` and removing the prepend of the full root path, see below. That at least works for me. However, it may not be the solution you want, as it is not clear to me, what the design considerations are and what would be sensible in golang.

```
diff --git a/cmd/add.go b/cmd/add.go
index d93517d..2e1e918 100644
--- a/cmd/add.go
+++ b/cmd/add.go
@@ -140,7 +140,7 @@ func addLocalHookDir(hookName string, fs afero.Fs) {
 }

 func getGitHooksDir() string {
-       cmd := exec.Command("sh", "-c", "git rev-parse --git-common-dir")
+       cmd := exec.Command("sh", "-c", "git rev-parse --absolute-git-dir")

        outputBytes, err := cmd.CombinedOutput()
        if err != nil {
@@ -148,7 +148,7 @@ func getGitHooksDir() string {
        }

        gitDir := strings.TrimSpace(string(outputBytes))
-       return filepath.Join(getRootPath(), gitDir, "hooks")
+       return filepath.Join(gitDir, "hooks")
 }

 func isLefthookFile(pathFile string) bool {
```

**Comment by jsmestad** [comment]
Ah, I think that looks OK to me assuming everything else works as intended.

**Comment by fahrinh** [comment]
@extmind 
Please make it as a PR. 
At least, I can fork it as a workaround until officially resolved.

**Comment by Envek** [comment]
Fix for this has been released in [0.7.5](https://github.com/evilmartians/lefthook/releases/tag/v0.7.5). Enjoy!
